README.md
# SwccSign
Southwestern Community College does own all logos, images, and intellectual property from this project. The project is currently used to to display custom visual information throughout the campus. 


PUBLICv1.01/changelog.txt
C:\xampp\htdocs\Php\thumbnails
http://localhost/Php/thumbnails/

Create Database signage;
Create table signage.signs (
Name varchar(255), 
Description varchar(255), 
Position varchar(255), 
Color varchar(255),
Size Integer(255),
ImageSize varchar(255),
Intervals Integer(255)
);

PUBLICv1.01/index.js
<script>
  function openInNewTab(url) {
  var win = window.open(url, '_blank');
  win.focus();
	}
</script>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
<script>
$('input').keypress(function(e) {
    if (e.which == 13) {
        $(this).next('input').focus();
        e.preventDefault();
    }
});
</script>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
<script>

var Source;
//This script allows the upload of an image and a preview of that image.
var imageLoader = document.getElementById('fileToUpload');
    imageLoader.addEventListener('change', handleImage, false);

function handleImage(e) {
    var reader = new FileReader();
    reader.onload = function (event) {
		Source=event.target.result;
        $('.uploader img').attr('src',event.target.result);
    }
    reader.readAsDataURL(e.target.files[0]);
}

</script>
<script>

var myWindow;

function openWin() {
myWindow = window.open("LiveDemo.html?"+ImageSize+"@"+FontSize+"@"+FontColor+"@"+FontPosition+"@"+Caption+"@"+Source, "", 'fullscreen=yes');
}

var output = document.getElementById("Output");
//Live Demo Command Divs
var livedemo = document.getElementById("LiveDemoWindow");
var otherdiv = document.getElementById("Collapseable");
//Demo Preview Div
var demoimage = document.getElementById("DemoImage");

//Image Size
var size = document.getElementById("ImageSize");
//Position
var position = document.getElementById("Position");
//Color
var color = document.getElementById("Color");
//Text
var text = document.getElementById("text");
//we neeed the double declaration, 1 for live 1 for generic
var slider = document.getElementById("Size");

//Live Demo Scripts
function CollapseWindow() {
if(otherdiv.style.display === "none") {
	otherdiv.style.display = "block";
} else {
    otherdiv.style.display = "none";
}
}

function Open() {
livedemo.style.display = "block";
}

function Exit() {
livedemo.style.display = "none";
}

var ImageSize = size.value;
var FontPosition = position.value;
var FontColor = color.value;
var FontSize = slider.value;
var Caption = text.value;
Update();
window.setInterval(function(){
  Update();
}, 2500);

function Update() {
if(Source) {
 demoimage.innerHTML = "<div class='container'><img src="+Source+" style="+ImageSize+"/>" + "<font size="+FontSize+"><font color="+FontColor+"><div class="+FontPosition+">" +Caption+"</font></font></div></div>";
 }
}

text.oninput = function() {
Caption = this.value;
}
//Image Size
size.oninput = function() {
ImageSize = this.value;
}
//Position
position.oninput = function() {
FontPosition = this.value;
}
//Color picker
color.oninput = function() {
  FontColor = this.value;
}

//Text size
slider.oninput = function() {
 output.innerHTML = this.value;
 FontSize = this.value;
}

</script>

<script>
function fade(evt, action) {
    var i, tabcontent, tablinks;
    tabcontent = document.getElementsByClassName("tabcontent");
    for (i = 0; i < tabcontent.length; i++) {
        tabcontent[i].style.display = "none";
    }
    tablinks = document.getElementsByClassName("tablinks");
    for (i = 0; i < tablinks.length; i++) {
        tablinks[i].className = tablinks[i].className.replace(" active", "");
    }
    document.getElementById(action).style.display = "block";
    evt.currentTarget.className += " active";
}
</script>

<script>
//Make the DIV element draggagle:
dragElement(document.getElementById(("LiveDemoWindow")));

function dragElement(elmnt) {
  var pos1 = 0, pos2 = 0, pos3 = 0, pos4 = 0;
  if (document.getElementById("LiveDemoHeader")) {
    /* if present, the header is where you move the DIV from:*/
    document.getElementById("LiveDemoHeader").onmousedown = dragMouseDown;
  } else {
    /* otherwise, move the DIV from anywhere inside the DIV:*/
    elmnt.onmousedown = dragMouseDown;
  }

  function dragMouseDown(e) {
    e = e || window.event;
    // get the mouse cursor position at startup:
    pos3 = e.clientX;
    pos4 = e.clientY;
    document.onmouseup = closeDragElement;
    // call a function whenever the cursor moves:
    document.onmousemove = elementDrag;
  }

  function elementDrag(e) {
    e = e || window.event;
    // calculate the new cursor position:
    pos1 = pos3 - e.clientX;
    pos2 = pos4 - e.clientY;
    pos3 = e.clientX;
    pos4 = e.clientY;
    // set the element's new position:
    elmnt.style.top = (elmnt.offsetTop - pos2) + "px";
    elmnt.style.left = (elmnt.offsetLeft - pos1) + "px";
  }

  function closeDragElement() {
    /* stop moving when mouse button is released:*/
    document.onmouseup = null;
    document.onmousemove = null;
  }
}
</script>

PUBLICv1.01/index.php
<title>Uploading Media</title>
<body>
<link rel="stylesheet" type="text/css" href="style.css">
<meta name="viewport" content="width=device-width, initial-scale=1">
<div class="container">
<?php
$file = 'C:\Xampp\htdocs\Php\index.php';
$bytes = readfile($file);

$target_dir = "thumbnails/";
$target_file = $target_dir . basename($_FILES["fileToUpload"]["name"]);
$uploadOk = 1;
$imageFileType = strtolower(pathinfo($target_file,PATHINFO_EXTENSION));
	$servername = "localhost";
	$username = "root";
	$password = "";
	$dbname = "signage";

	$conn = new mysqli($servername, $username, $password, $dbname);
	if ($conn->connect_error){
	die("Connection failed: " . $sconn->connect_error);
	}

	$submit = $_POST['submit'];
	if(isset($submit)) {
// Check if image file is a actual image or fake image
    $check = getimagesize($_FILES["fileToUpload"]["tmp_name"]);
    if($check !== false) {
	//We needed to set this to 1 but didn't know how. For now we will just set it to 1 and then catch everything else in $imageFileType.....
    $uploadOk = 1;
    } else {
        $uploadOk = 1;
    }
// Check if file already exists
if (file_exists($target_file)) {
    echo "Sorry, file already exists.";
    $uploadOk = 0;
}
// Allow certain file formats
if($imageFileType != "jpg" && $imageFileType != "png" && $imageFileType != "jpeg"
&& $imageFileType != "gif" && $imageFileType != "mov" && $imageFileType != "mp4" && $imageFileType != "webm") {
    echo "Sorry, only webm, MOV, MP4, JPG, JPEG, PNG & GIF files are allowed.";
    $uploadOk = 0;
}
// Check if $uploadOk is set to 0 by an error
if ($uploadOk == 0) {
    echo "Sorry, your file was not uploaded.";
// if everything is ok, try to upload file
} else {
    if (move_uploaded_file($_FILES["fileToUpload"]["tmp_name"], $target_file)) {
        echo "The file ". basename( $_FILES["fileToUpload"]["name"]). " has been uploaded.";
    } else {
        echo "Sorry, there was an error uploading your file.";
    }
	$Name = $_FILES["fileToUpload"]["name"];
	$Description = $_POST['Description'];
	$Position = $_POST['Position'];
	$Color = $_POST['Color'];
	$Size = $_POST['Size'];
	$ImageSize = $_POST['ImageSize'];
	$Intervals = $_POST['Intervals'];

	if($ImageSize=="video") {
	$Description='Video';
	$Position = 'Video';
	$Color = 'Video';
	}
	
	$sql = "INSERT INTO signs (Name, Description, Position, Color,Size,ImageSize,Intervals) VALUES ('$Name','$Description','$Position','$Color','$Size','$ImageSize','$Intervals')";
	
	if($conn->query($sql) === TRUE) {
	echo "<script type='text/javascript'>alert('Inquiry saved.')</script>";
	echo '<script type="text/javascript"> window.location = "http://localhost/Php/index.html"</script>';
	} else {
	echo "<script type='text/javascript'>alert('There was an error. Image not saved.')</script>";
	echo "Error";
	}
	$conn->close();
}
	}
?>

</div>
</body>
<div class="version">
Version 2.6.106, Eian / Laura / Morgan
</div>

PUBLICv1.01/delete.php
<link rel="stylesheet" type="text/css" href="delete.css">
<body>
<title>Slideshow Preview</title>
<div class="topright">
<img src="Images/logo.png" width="193" height="77"></img>
</div>

<div class="tab">
  <button class="tablinks" onclick="window.location.href='index.html'">Link Information</button>
  <button class="tablinks" onclick="window.location.href='Display.php'">Display Webpage</button>
  <button class="tablinks" onclick="window.location.href='delete.php'">Preview Images And Captions</button>
</div>

<?php
define ('IMGDIR', 'C:\xampp\htdocs\Php\thumbnails');
define ('WEBIMGDIR', 'http://localhost/Php/thumbnails/');
$err = '';
$ss = new slideshow($err);
if (($err = $ss->init()) != ''){
	header('HTTP/1.1 500 Internal Server Error');
	echo $err;
	exit();
}
$ss->get_images();
list($curr, $caption, $first, $prev, $next, $last) = $ss->run();
class slideshow {
	private $files_arr = NULL;
	private $err = NULL;
	public function __construct(&$err){
		$this->files_arr = array();
		$this->err = $err;
	}
	public function init(){

		if (!$this->dir_exists())
		{
			return 'Error retrieving images, missing directory';
		}
		return '';
	}
	public function get_images(){
		if (isset($_SESSION['imgarr'])){
			$this->files_arr = $_SESSION['imgarr'];
		} else {
			if ($dh = opendir(IMGDIR)){
				while (false !== ($file = readdir($dh))){
					if (preg_match('/^.*\.(jpg|jpeg|gif|png)$/i', $file)){
						$this->files_arr[] = $file;
					}
				}
				closedir($dh);
			}
			$_SESSION['imgarr'] = $this->files_arr;
		}
	}
	public function run(){
		$curr = 1;
		$last = count($this->files_arr);
		if (isset($_GET['img'])){
			if (preg_match('/^[0-9]+$/', $_GET['img'])) $curr = (int)  $_GET['img'];
			if ($curr <= 0 || $curr > $last) $curr = 1;
		}
		if ($curr <= 1) {
			$prev = $curr;
			$next = $curr + 1;
		} else if ($curr >= $last) {
			$prev = $last - 1;
			$next = $last;
		} else {
			$prev = $curr - 1;
			$next = $curr + 1;
		}
	$IMG = array();
	$CAP = array();
	$FORMAT = array();
	$caption = str_replace('-', ' ', $this->files_arr[$curr - 1]);
	$caption = str_replace('_', ' ', $caption);
	//foreach ($this->files_arr as $value) {
	$connection = mysqli_connect("localhost","root","","signage");
    $result = mysqli_query($connection,"SELECT * FROM signs WHERE 1");
    while($row = mysqli_fetch_array($result)){
	$Description = $row['Description'];
	$Color = $row['Color'];
	$Name = $row['Name'];//
	$Position = $row['Position'];
	$Size = $row['Size'];
	$ImageSize = $row['ImageSize'];//
	array_push($IMG,[$Name, $ImageSize],[$ImageSize,$Name]);
	array_push($FORMAT,[$Position, $Color],[$Color,$Position]);
	array_push($CAP,[$Description, $Size],[$Size,$Description]);
	}
$Counter=0;
//$a Is $Name and $b is $ImageSize

foreach ($IMG as list($a,$b)) {
	if($a==null) {
	} else {
	if($b=="video") {
	echo "<div class='container'>";
	echo "<video controls>";
	echo "<source src='thumbnails/$a'></source>";
	echo "</video>";
	echo "<form action='sendto.php' method='POST'>";
	echo "<input type='checkbox' id='Delete' name='Name' value=$a></input>";
	echo "<label for='Delete'>Delete Video?</label>";
	echo "<div id=$Counter></div></div>";
	} else {
	if (in_array($a, $this->files_arr)) {
	$Counter=$Counter+1;
	echo "<div class='container'><img src='thumbnails/$a' style='$b'>";
	echo "<form action='sendto.php' method='POST'>";
	echo "<input type='checkbox' id='Delete' name='Name' value=$a></input>";
	echo "<label for='Delete'>Delete Image?</label>";
	echo "<div id=$Counter></div>";
	echo "</div>";
	}
	}
	}
}
	$Counter2=0;
	foreach ($FORMAT as list($a,$b)) {
	foreach ($CAP as list ($c,$d)) {
	$Counter2=$Counter2+1;
	//$A is Position $b is color $c is text size $d is description
	//echo "<div class='container'><img src='thumbnails/$value' style='$ImageSize'>";
	//echo "<div class='$Position'><label>(*$Position*)</label><font color='$Color'><font size='$Size'>$Description</font></font></div>";
	if($Counter2<=$last) {
	echo "<div class=$b><label>(*$b*)</label><font color=$a><font size=$c>$d</font></font></div></div>";
	}
	}
	}
	?>
	
	</div>
	</div>
	<div class="Collapseable">
	<input type="submit" value="Delete Images" name="submit" onClick="Send();"></input>
	<input type="reset" value="Undo Selections" name="reset"></input>
	</form>
	<div class="version">
	Version 2.6.106, Eian / Laura / Morgan
	</div>
	</div>
	<?php
	return array($this->files_arr[$curr - 1], "", 1, $prev, $next, $last);
	}
	
	private function dir_exists(){
		return file_exists(IMGDIR);
	}
}
?>
<script>

for(i=1; i<=<?php echo $last; ?>; i++) {
x = i + "";
document.getElementById(x+"").innerHTML = <?php echo $; ?>;
}

function Send() {
window.location = "http://localhost/Php/delete.php";
}
</script>
</form>
</div>
</body>

</html>

PUBLICv1.01/sendto.php
<!DOCTYPE html>
<html>
<title>Unlinking the image.</title>
<body>

<link rel="stylesheet" type="text/css" href="sytle.css">
<div class="container">
<?php
$file = 'C:\Xampp\htdocs\Php\delete.php';
$bytes = readfile($file);

$servername = "localhost";
$username = "root";
$password = "";
$dbname = "signage";

$conn = new mysqli($servername, $username, $password, $dbname);
if ($conn->connect_error){
	die("Connection failed: " . $sconn->connect_error);
}
$submit = $_POST['submit'];
$Name = $_POST['Name'];

if(isset($_POST['Name']) && $_POST['Name'] == $Name){
	$sql = "DELETE FROM `signs` WHERE `Name`='$Name'";
	if($conn->query($sql) === TRUE) {
	unlink('thumbnails/'."$Name");
	echo("Successfully deleted "."$Name");
	echo '<script type="text/javascript"> alert("Successfully deleted "."$Name");</script>';
	} else {
   echo "Error!";
	}
	$conn->close();
} else{
  echo "Error Image not found.";
}
?>
</div>
</body>
</html>

PUBLICv1.01/Scroller.js
;window.Modernizr=function(a,b,c){function x(a){j.cssText=a}function y(a,b){return x(prefixes.join(a+";")+(b||""))}function z(a,b){return typeof a===b}function A(a,b){return!!~(""+a).indexOf(b)}function B(a,b){for(var d in a)if(j[a[d]]!==c)return b=="pfx"?a[d]:!0;return!1}function C(a,b,d){for(var e in a){var f=b[a[e]];if(f!==c)return d===!1?a[e]:z(f,"function")?f.bind(d||b):f}return!1}function D(a,b,c){var d=a.charAt(0).toUpperCase()+a.substr(1),e=(a+" "+n.join(d+" ")+d).split(" ");return z(b,"string")||z(b,"undefined")?B(e,b):(e=(a+" "+o.join(d+" ")+d).split(" "),C(e,b,c))}var d="2.5.3",e={},f=!0,g=b.documentElement,h="modernizr",i=b.createElement(h),j=i.style,k,l={}.toString,m="Webkit Moz O ms",n=m.split(" "),o=m.toLowerCase().split(" "),p={},q={},r={},s=[],t=s.slice,u,v={}.hasOwnProperty,w;!z(v,"undefined")&&!z(v.call,"undefined")?w=function(a,b){return v.call(a,b)}:w=function(a,b){return b in a&&z(a.constructor.prototype[b],"undefined")},Function.prototype.bind||(Function.prototype.bind=function(b){var c=this;if(typeof c!="function")throw new TypeError;var d=t.call(arguments,1),e=function(){if(this instanceof e){var a=function(){};a.prototype=c.prototype;var f=new a,g=c.apply(f,d.concat(t.call(arguments)));return Object(g)===g?g:f}return c.apply(b,d.concat(t.call(arguments)))};return e}),p.cssanimations=function(){return D("animationName")};for(var E in p)w(p,E)&&(u=E.toLowerCase(),e[u]=p[E](),s.push((e[u]?"":"no-")+u));return x(""),i=k=null,function(a,b){function g(a,b){var c=a.createElement("p"),d=a.getElementsByTagName("head")[0]||a.documentElement;return c.innerHTML="x<style>"+b+"</style>",d.insertBefore(c.lastChild,d.firstChild)}function h(){var a=k.elements;return typeof a=="string"?a.split(" "):a}function i(a){var b={},c=a.createElement,e=a.createDocumentFragment,f=e();a.createElement=function(a){var e=(b[a]||(b[a]=c(a))).cloneNode();return k.shivMethods&&e.canHaveChildren&&!d.test(a)?f.appendChild(e):e},a.createDocumentFragment=Function("h,f","return function(){var n=f.cloneNode(),c=n.createElement;h.shivMethods&&("+h().join().replace(/\w+/g,function(a){return b[a]=c(a),f.createElement(a),'c("'+a+'")'})+");return n}")(k,f)}function j(a){var b;return a.documentShived?a:(k.shivCSS&&!e&&(b=!!g(a,"article,aside,details,figcaption,figure,footer,header,hgroup,nav,section{display:block}audio{display:none}canvas,video{display:inline-block;*display:inline;*zoom:1}[hidden]{display:none}audio[controls]{display:inline-block;*display:inline;*zoom:1}mark{background:#FF0;color:#000}")),f||(b=!i(a)),b&&(a.documentShived=b),a)}var c=a.html5||{},d=/^<|^(?:button|form|map|select|textarea)$/i,e,f;(function(){var a=b.createElement("a");a.innerHTML="<xyz></xyz>",e="hidden"in a,f=a.childNodes.length==1||function(){try{b.createElement("a")}catch(a){return!0}var c=b.createDocumentFragment();return typeof c.cloneNode=="undefined"||typeof c.createDocumentFragment=="undefined"||typeof c.createElement=="undefined"}()})();var k={elements:c.elements||"abbr article aside audio bdi canvas data datalist details figcaption figure footer header hgroup mark meter nav output progress section summary time video",shivCSS:c.shivCSS!==!1,shivMethods:c.shivMethods!==!1,type:"default",shivDocument:j};a.html5=k,j(b)}(this,b),e._version=d,e._domPrefixes=o,e._cssomPrefixes=n,e.testProp=function(a){return B([a])},e.testAllProps=D,g.className=g.className.replace(/(^|\s)no-js(\s|$)/,"$1$2")+(f?" js "+s.join(" "):""),e}(this,this.document),function(a,b,c){function d(a){return o.call(a)=="[object Function]"}function e(a){return typeof a=="string"}function f(){}function g(a){return!a||a=="loaded"||a=="complete"||a=="uninitialized"}function h(){var a=p.shift();q=1,a?a.t?m(function(){(a.t=="c"?B.injectCss:B.injectJs)(a.s,0,a.a,a.x,a.e,1)},0):(a(),h()):q=0}function i(a,c,d,e,f,i,j){function k(b){if(!o&&g(l.readyState)&&(u.r=o=1,!q&&h(),l.onload=l.onreadystatechange=null,b)){a!="img"&&m(function(){t.removeChild(l)},50);for(var d in y[c])y[c].hasOwnProperty(d)&&y[c][d].onload()}}var j=j||B.errorTimeout,l={},o=0,r=0,u={t:d,s:c,e:f,a:i,x:j};y[c]===1&&(r=1,y[c]=[],l=b.createElement(a)),a=="object"?l.data=c:(l.src=c,l.type=a),l.width=l.height="0",l.onerror=l.onload=l.onreadystatechange=function(){k.call(this,r)},p.splice(e,0,u),a!="img"&&(r||y[c]===2?(t.insertBefore(l,s?null:n),m(k,j)):y[c].push(l))}function j(a,b,c,d,f){return q=0,b=b||"j",e(a)?i(b=="c"?v:u,a,b,this.i++,c,d,f):(p.splice(this.i++,0,a),p.length==1&&h()),this}function k(){var a=B;return a.loader={load:j,i:0},a}var l=b.documentElement,m=a.setTimeout,n=b.getElementsByTagName("script")[0],o={}.toString,p=[],q=0,r="MozAppearance"in l.style,s=r&&!!b.createRange().compareNode,t=s?l:n.parentNode,l=a.opera&&o.call(a.opera)=="[object Opera]",l=!!b.attachEvent&&!l,u=r?"object":l?"script":"img",v=l?"script":u,w=Array.isArray||function(a){return o.call(a)=="[object Array]"},x=[],y={},z={timeout:function(a,b){return b.length&&(a.timeout=b[0]),a}},A,B;B=function(a){function b(a){var a=a.split("!"),b=x.length,c=a.pop(),d=a.length,c={url:c,origUrl:c,prefixes:a},e,f,g;for(f=0;f<d;f++)g=a[f].split("="),(e=z[g.shift()])&&(c=e(c,g));for(f=0;f<b;f++)c=x[f](c);return c}function g(a,e,f,g,i){var j=b(a),l=j.autoCallback;j.url.split(".").pop().split("?").shift(),j.bypass||(e&&(e=d(e)?e:e[a]||e[g]||e[a.split("/").pop().split("?")[0]]||h),j.instead?j.instead(a,e,f,g,i):(y[j.url]?j.noexec=!0:y[j.url]=1,f.load(j.url,j.forceCSS||!j.forceJS&&"css"==j.url.split(".").pop().split("?").shift()?"c":c,j.noexec,j.attrs,j.timeout),(d(e)||d(l))&&f.load(function(){k(),e&&e(j.origUrl,i,g),l&&l(j.origUrl,i,g),y[j.url]=2})))}function i(a,b){function c(a,c){if(a){if(e(a))c||(j=function(){var a=[].slice.call(arguments);k.apply(this,a),l()}),g(a,j,b,0,h);else if(Object(a)===a)for(n in m=function(){var b=0,c;for(c in a)a.hasOwnProperty(c)&&b++;return b}(),a)a.hasOwnProperty(n)&&(!c&&!--m&&(d(j)?j=function(){var a=[].slice.call(arguments);k.apply(this,a),l()}:j[n]=function(a){return function(){var b=[].slice.call(arguments);a&&a.apply(this,b),l()}}(k[n])),g(a[n],j,b,n,h))}else!c&&l()}var h=!!a.test,i=a.load||a.both,j=a.callback||f,k=j,l=a.complete||f,m,n;c(h?a.yep:a.nope,!!i),i&&c(i)}var j,l,m=this.yepnope.loader;if(e(a))g(a,0,m,0);else if(w(a))for(j=0;j<a.length;j++)l=a[j],e(l)?g(l,0,m,0):w(l)?B(l):Object(l)===l&&i(l,m);else Object(a)===a&&i(a,m)},B.addPrefix=function(a,b){z[a]=b},B.addFilter=function(a){x.push(a)},B.errorTimeout=1e4,b.readyState==null&&b.addEventListener&&(b.readyState="loading",b.addEventListener("DOMContentLoaded",A=function(){b.removeEventListener("DOMContentLoaded",A,0),b.readyState="complete"},0)),a.yepnope=k(),a.yepnope.executeStack=h,a.yepnope.injectJs=function(a,c,d,e,i,j){var k=b.createElement("script"),l,o,e=e||B.errorTimeout;k.src=a;for(o in d)k.setAttribute(o,d[o]);c=j?h:c||f,k.onreadystatechange=k.onload=function(){!l&&g(k.readyState)&&(l=1,c(),k.onload=k.onreadystatechange=null)},m(function(){l||(l=1,c(1))},e),i?k.onload():n.parentNode.insertBefore(k,n)},a.yepnope.injectCss=function(a,c,d,e,g,i){var e=b.createElement("link"),j,c=i?h:c||f;e.href=a,e.rel="stylesheet",e.type="text/css";for(j in d)e.setAttribute(j,d[j]);g||(n.parentNode.insertBefore(e,n),m(c,0))}}(this,document),Modernizr.load=function(){yepnope.apply(window,[].slice.call(arguments,0))};

PUBLICv1.01/Display.php
<link rel="stylesheet" type="text/css" href="display.css">
<style>
*{margin:0;}

html, body{
    height: 100%;
    overflow:hidden;
}
</style>
<?php
header("Expires: Tue, 01 Jan 2000 00:00:00 GMT");
header("Last-Modified: " . gmdate("D, d M Y H:i:s") . " GMT");
header("Cache-Control: no-store, no-cache, must-revalidate, max-age=0");
header("Cache-Control: post-check=0, pre-check=0", false);
header("Pragma: no-cache");

define ('IMGDIR', 'C:\xampp\htdocs\Php\thumbnails');
define ('WEBIMGDIR', 'http://localhost/Php/thumbnails/');
$err = '';

// init slideshow class
$ss = new slideshow($err);
if (($err = $ss->init()) != '')
{
	header('HTTP/1.1 500 Internal Server Error');
	echo $err;
	exit();
}
$ss->get_images();
list($curr, $caption, $first, $prev, $next, $last) = $ss->run();

class slideshow{
	private $files_arr = NULL;
	private $err = NULL;
	
	public function __construct(&$err){
		$this->files_arr = array();
		$this->err = $err;
	}
	
	public function init(){
		if (!$this->dir_exists()){
			return 'Error retrieving images, missing directory';
		}
		return '';
	}

	public function get_images(){
		if (isset($_SESSION['imgarr'])){
			$this->files_arr = $_SESSION['imgarr'];
		} else {
			if ($dh = opendir(IMGDIR)){
				while (false !== ($file = readdir($dh))){
					if (preg_match('/^.*\.(jpg|jpeg|gif|png|mov|mp4|webm)$/i', $file)){
						$this->files_arr[] = $file;
					}
				}
				closedir($dh);
			}
			$_SESSION['imgarr'] = $this->files_arr;
		}
	}
	
	public function run(){
		$curr = 1;
		$last = count($this->files_arr);
		$GlobalSeconds=10000;
		if (isset($_GET['img'])){
			if (preg_match('/^[0-9]+$/', $_GET['img'])) $curr = (int)  $_GET['img'];
			if ($curr <= 0 || $curr > $last) $curr = 1;
		} 
		if ($curr <= 1){
			$prev = $curr;
			$next = $curr + 1;
		} else if ($curr >= $last) {
			$prev = $last - 1;
			$next = $last;
		} else {
			$prev = $curr - 1;
			$next = $curr + 1;
		}
	$caption = str_replace('-', ' ', $this->files_arr[$curr - 1]);
	$caption = str_replace('_', ' ', $caption);
	
	$connection = mysqli_connect("localhost","root","","signage"); 
    $result = mysqli_query($connection,"SELECT * FROM signs WHERE Name='$caption'");
	if($result!=null) {
    while($row = mysqli_fetch_array($result)){
	$Description = $row['Description'];
	$Color = $row['Color'];
	$Name = $row['Name'];
	$Position = $row['Position'];
	$Size = $row['Size'];
	$ImageSize = $row['ImageSize'];
	$Intervals = $row['Intervals'];
	}
	?>
	<script>
	var last = <?php echo $last ?>;
	var test = location;
	var counter = test.search.replace("?img=",'');
	var x = setInterval(function() {
	//Change the image.
	counter++;
	if (counter <= last) {
	var setto = "http://localhost/Php/Display.php?img="+counter;
	if(setto!=location) {
		window.location = setto;
	} else console.error("ERROR");
	} else if(counter >= last) {
		window.location = "http://localhost/Php/Display.php?img=1";
	} else {
	clearInterval(x);
	}
	}, <?php echo $Intervals *1000;?>);
</script>
	<?php
	if($Description=="Video") {
	echo "<video width='100%' height='100%' autoplay allowfullscreen>";
	echo "<source src='thumbnails/$Name'></source>";
	echo "</video>";
	} else {
	echo "<div class='container'>";
	echo "<div class='gallery fade'>";
	echo "<img src='thumbnails/$Name' style='$ImageSize'>";
	echo "</div>";
	if($Position!=null) {
		echo "<div class='$Position'><font color='$Color'><font size='$Size'>$Description</div></div>";
		echo "</div></div>";
	} else echo "Error!";
	}
	
	return array($this->files_arr[$curr - 1], "", 1, $prev, $next, $last);
	}
	}
	
	private function dir_exists()
	{
		return file_exists(IMGDIR);
	}
}
?>

<body>
	<div id="gallery">
        <p><?=$caption;?></p>
    </div>
</body>
<div class="version">
Version 2.6.106, Eian / Laura / Morgan
</div>
</html>
